docs/configuration.md

```diff
@@ -0,0 +1,158 @@
+
+    ```ruby
+    #!config/initializers/wallaby.rb
+    Wallaby.config do |config|
+      config.security.authenticate do
+        # controller methods can be used inside this block
+        authenticate_or_request_with_http_basic do |username, password|
+          username == 'too_simple' && password == 'too_naive'
+        end
+      end
+
+      config.security.current_user do
+        # controller methods can be used inside this block
+        Class.new do
+          # email field is for gravator profile image
+          def email
+            'user@example.com'
+          end
+        end.new
+      end
+
+      # This will be used to retrive the email from current_user above to show a gravatar image.
+      config.security.email_method = :email
+    end
+    ```
+
+### Logging Out
+
+Wallaby doesn't do log-out function. However, it provides two options to send user to the logout route:
+
+- `logout_path`: the name of the logout url helper.
+- `logout_method`: the HTTP request verb for this logout url.
+
+Given the following route:
+
+```ruby
+#!config/routes.rb
+Rails.application.routes.draw do
+  delete 'logout', to: 'sessions#destroy', as: :logging_out
+end
+```
+
+Then configuration should go:
+
+```ruby
+#!config/initializers/wallaby.rb
+Wallaby.config do |c|
+  c.security.logout_path = 'logging_out_path'
+  # or
+  c.security.logout_path = 'logging_out_url'
+  c.security.logout_method = 'delete'
+end
+```
+
+### Authorization
+
+Wallaby supports Cancancan, please see its [documentation](https://github.com/CanCanCommunity/cancancan/wiki) to find out how to set up authorization to restrict end-users from accessing/editing/deleting a model.
+
+### Models
+
+What models Wallaby should be handling with can be configured in the following two ways:
+
+1. Blacklist the models that Wallaby shouldn't even display:
+
+    ```ruby
+    #!config/initializers/wallaby.rb
+    Wallaby.config do |config|
+      config.models.exclude User, Organization
+      # or
+      # config.models.exclude 'User', 'Organization'
+      # or
+      # config.models.exclude [User, Organization]
+      # or
+      # config.models.exclude ['User', 'Organization']
+    end
+    ```
+
+2. Whitelist the models that Wallaby should take care of (Note that once the white list is defined, Wallaby will ignore the black list above.):
+
+    ```ruby
+    #!config/initializers/wallaby.rb
+    Wallaby.config do |config|
+      config.models = [User, Organization]
+      # or
+      # config.models = 'User', 'Organization'
+      # or
+      # config.models = [User, Organization]
+      # or
+      # config.models = ['User', 'Organization']
+    end
+    ```
+
+### Metadata
+
+Be able to configure global metadata that Wallaby should use as default value when decorator-defined metadata is absent.
+
+- For now, it is possible to configure the max length that Wallaby should truncate when given text is longer than defined:
```

Comment:
Could you provide an overview of the other metadata that can be configured as well?

Reply:
it's the only option for now :(